airflow/sensors/time_sensor.py

```diff
@@ -72,13 +73,8 @@ def __init__(self, *, target_time: datetime.time, **kwargs) -> None:
 
         self.target_datetime = timezone.convert_to_utc(aware_time)
 
-    def execute(self, context: Context) -> NoReturn:
-        trigger = DateTimeTrigger(moment=self.target_datetime)
+    def execute(self, context: Context) -> None:
         self.defer(
-            trigger=trigger,
-            method_name="execute_complete",
+            method_name=TaskDeferred.TRIGGER_EXIT,
+            trigger=DateTimeTrigger(moment=self.target_datetime, end_task=True),
```

Comment:
If we already set `method_name` to `TaskDeferred.TRIGGER_EXIT`, what's the propose of `end_task`?

Reply:
I have removed `TaskDeferred.TRIGGER_EXIT` as it was sentinel value to signal the expectation that the trigger will exit the task. If the trigger will terminate the task instance itself, `method_name` should be None. Otherwise, provide the name of the method to be used when resuming execution in the task.